### PR TITLE
Add end_key to kv scan request

### PR DESCRIPTION
**Pull request by MyonKeminta**
Signed-off-by: MyonKeminta <MyonKeminta@users.noreply.github.com>

### CODE REVIEW

File: proto/kvrpcpb.proto [hunk 1]
```diff
@@ -98,6 +98,10 @@ message ScanRequest {
     uint64 version = 4;
     bool key_only = 5;
     bool reverse = 6;
+    // For compatibility, when scanning forward, `start_key` is the inclusive lower_bound and `end_key`
+    // is the exclusive upper_bound; when scanning backward, `start_key` is the exclusive upper_bound
+    // and `end_key` is the inclusive lower_bound.
+    bytes end_key = 7;
```
Comment (siddontang) [inline proto/kvrpcpb.proto]: maybe we can use `[start_key, end_key)` or `[end_key, start_key)` for simple explanation. 

But, for backward, maybe start_key is inclusive?

Comment (MyonKeminta) [reply proto/kvrpcpb.proto]: In the current implementation in TiKV, it's exclusive.

**Comment by MyonKeminta** [comment]
@siddontang PTAL again @zhangjinpeng1987 PTAL

File: proto/kvrpcpb.proto (hunk 1 above)
Comment (zhangjinpeng1987) [reply proto/kvrpcpb.proto]: Seems we don't use reverse scan?

**Comment by MyonKeminta** [comment]
@siddontang PING

**Pull request merged**